kubric/randomness.py

```diff
@@ -67,17 +67,8 @@ def position_sampler(region):
   region = np.array(region, dtype=np.float)
 
   def _sampler(obj: objects.PhysicalObject, rng):
```

Comment:
with `rng.uniform` this method would only be corrected for a **cube** object of **unit** size, no?
For other geometries you need rejection sampling (or other tricks people use in rendering).

Even if this is an internal method, shouldn't this be documented?

Reply:
This line: `effective_region = np.array(region) - obj.aabbox` makes sure that even scaled and rotated objects are fully within the region (because the `aabbox` property takes both into account).
It is a conservative approach because it relies on scaling and rotating the `bounds` and then using the smallest axis-aligned bbox that encloses the transformed bounds. We could do something more sophisticated to allow objects to be placed even closer to the boundaries, but I don't think that's worth it.